### PR TITLE
Docs: Update parser-cxx.rst

### PR DESCRIPTION
**Pull request by pragmaware**

### CODE REVIEW

File: docs/parser-cxx.rst [hunk 1]
```diff
@@ -31,11 +31,151 @@ features were added. Among them:
...
+	DECLARE_FUNCTION(b)
+
+will be processed as
+
+.. code-block:: C
+
+	int aCall();
+	int bCall();
+
+Macros with variable arguments use the gcc __VA_ARGS__ syntax.
```
Comment (b4n) [inline docs/parser-cxx.rst]: might be worth wrapping `__VA_ARGS__` in ``es to avoid it being treated as underline

Comment (pragmaware) [inline docs/parser-cxx.rst]: Done.

**Comment by masatake** [comment]
@pragmaware, could you allow me to advise about git usage?

When adding extra small commits for fixing exiting commits, `git rebase --autosquash` is helpfu.
See http://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html

**Comment by pragmaware** [comment]
@masatake, but can I do a --fixup even if I have already pushed my previous commit to a remote repo?

**Comment by masatake** [comment]
It should be done in the final stage. During these discussions and development, you don't have to squash at all.

What you need in these stages is to write "fixup!" or "squash!" in the commit log of the extra small commits. When merging to the upstream, I called this "final stage" at the head of this comment, you can do `rebase --autosquash` and the do `push --force`. So the upstream can be kept simple and clean.
Do I make sense?

**Comment by pragmaware** [comment]
Hm.. I have always merged via github UI. I'll try commandline then.

**Comment by pragmaware** [comment]
OK. So I have pushed https://github.com/universal-ctags/ctags/commit/c775c46f3a73c76e8d884a532a18ef2e0ecbb693 via commandline after using fixup to merge the two commits. This worked great. Now I just close this pull request manually, right?

**Comment by masatake** [comment]
If you are using improve-cxx-parser-documentation, you can do push --force to your remote repository at git hub. The you can push "Merge pull request". You can use github UI.

Again, I will write typical work flow.

1. you push a branch to your remote repo.
2. you make a pull request.
3. someone gives you comments.
4. you add commits on your branch. Some of them may have "fixup!" in their commit log.
5. all reviewing and test on CI are passed. Now ready to merge.
6. you do "git rebase -i --autosquash" to build clean and simplified history.
7. you do "git push --force" to the branch.
8. CI runs again. and may says O.K.
9. you push the green "Merge pull request" button.

**Comment by pragmaware** [comment]
Aha.. OK. Will try this next time. Thans for the explaination :)
I'll close this now as I have already merged the changes.

**Pull request closed without merging**